vm/string.go

```diff
@@ -117,10 +133,11 @@ func builtinStringInstanceMethods() []*BuiltInMethodObject {
 				return func(t *thread, args []Object, blockFrame *callFrame) Object {
 
 					leftValue := receiver.(*StringObject).Value
-					right, ok := args[0].(*StringObject)
+					r := args[0]
+					right, ok := r.(*StringObject)
 
 					if !ok {
-						return wrongTypeError(receiver.Class())
+						return initErrorObject(TypeErrorClass, "Expect argument to be String. got=%T", r)
```

Comment:
You shouldn't use `%T` because that prints object's type in Go, which might be `*IntegerObject` but what we want is `Integer`. So just use `%s` and `r.Class().Name`.